lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3OutputStream.java

```diff
@@ -406,4 +466,20 @@ private void abortUploadSuppressed(Throwable throwable)
             }
         }
     }
+
+    private Tagging trinoCreationTag()
+    {
+        return Tagging.builder()
+                .tagSet(Tag.builder()
+                        .key(TRINO_CREATION_TAG)
+                        .value(trinoCompletionRequestId)
+                        .build())
+                .build();
+    }
+
+    private Predicate<Tag> isTrinoCreationTag()
```

Comment:
nit: better name would be `hasOurTrinoCreationTag`

Reply:
it's a predicate on a tag, not object, so "has..." is not appropriate.